homeassistant/components/wemo/light.py

```diff
@@ -141,16 +138,18 @@ def color_temp(self) -> int | None:
         """Return the color temperature of this light in mireds."""
         return cast(Optional[int], self.light.state.get("temperature_mireds"))
 
+    @property
+    def color_mode(self) -> ColorMode:
```

Comment:
@esev is this the correct way to detect the ColorMode?
Do all lights support either HS or COLOR_TEMP, or do some lights only support brightness or even only on/off?

Reply:
I think this is fine as it does not change the behavior.
Let's give @esev a week before merging though.

Reply:
Sorry for the delay in responding. It's possible to have on/off and brightness only lights too.

There is a self.light.capabilities list that contains the features of the device. I believe the mapping should be:

"onoff": ColorMode.ONOFF
"levelcontrol": ColorMode.BRIGHTNESS
"colorcontrol": ColorMode.XY
"colortemperature": ColorMode.COLOR_TEMP

This list is created [here](https://github.com/pywemo/pywemo/blob/bec12ac573870b208b823078e010ade9888de3a2/pywemo/ouimeaux_device/bridge.py#L362) using the names from [CAPABILITY_ID2NAME](https://github.com/pywemo/pywemo/blob/bec12ac573870b208b823078e010ade9888de3a2/pywemo/ouimeaux_device/bridge.py#L16)

Maybe this can be changed to:

```
if "colorcontrol" in self.light.capabilities:
    return ColorMode.HS
if "colortemperature" in self.light.capabilities:
    return ColorMode.COLOR_TEMP
if "levelcontrol" in self.light.capabilities:
    return ColorMode.BRIGHTNESS
return ColorMode.ONOFF
```

I'm not sure why the current implementation of this integration specifies it supports COLOR_MODE_HS, but then just converts that to COLOR_MODE_XY inside the turn_on method; that predates my involvement. Could we change that in this PR, and use ColorMode.XY when the "colorcontrol" is in the capabilities list?

Reply:
FWIW, I have a light that supports "colorcontrol". What's the best way for me to test this PR? Are there any frontend changes that would be needed?

Reply:
I have adjusted this as per your recommendation. I have also adjusted the `supported_colour_modes` accordingly.

You can test it straight away, as the frontend changes have been available for about a year already (the deprecation PR was back in March 2021: #47720).

Reply:
> Could we change that in this PR, and use ColorMode.XY when "colorcontrol" is in the capabilities list?

done